registry/config.go

```diff
@@ -207,7 +232,16 @@ func ValidateIndexName(val string) (string, error) {
 	if strings.HasPrefix(val, "-") || strings.HasSuffix(val, "-") {
 		return "", fmt.Errorf("Invalid index name (%s). Cannot begin or end with a hyphen.", val)
 	}
-	// *TODO: Check if valid hostname[:port]/ip[:port]?
+
+	parsedURL, err := url.Parse(val)
+	if err != nil {
+		return "", fmt.Errorf("Invalid index url (%s): %v\n", val, err)
+	}
+
+	if parsedURL.Host != "" {
```

Comment:
This only has the effect of stripping the scheme from the URL. For most index name this will not be parsed correctly. See https://play.golang.org/p/mpu-2C7n9F. This function should not be responsible for sanitizing the index name, the "index.docker.io -> docker.io" is an unfortunate hack but should not make changes for any private registry.